src/kernel/src/sys/syscall.c

```diff
@@ -0,0 +1,81 @@
+#include "syscall.h"
+#include "stdio.h"
+
+// if error, return -1
+int syscall(int sys, ...) {
+    va_list args;
+    va_start(args, sys);
+
+	// variables that can be passed to the SWI
+    int r0 = 0, r1 = 0, r2 = 0, r3 = 0;
+
+    // TODO: the interrupt handler should be able to get the pid from the scheduler?
+    switch(sys) {
+        case SYS_fork:
```

Comment:
awesome! this looks like a great starting set of syscalls

Reply:
Great! We can discuss if we need more in the next meeting.

Reply:
How did you choose the syscall numbers? Did you just make them equal to the posix counterpart?

Reply:
For most of them I used the values defined in `interrupt.h`. If the syscalls were not  present there, I used the Linux counterparts.